src/functional/iterator/embedded.py

```diff
@@ -245,18 +317,20 @@ def group_offsets(*offsets):
     return complete_offsets, tag_stack
 
 
-def shift_position(pos, *complete_offsets, offset_provider):
+def shift_position(
+    pos: MaybePosition, *complete_offsets: tuple[str, int], offset_provider: OffsetProvider
+) -> MaybePosition:
     if pos is None:
         return None
-    new_pos = pos.copy()
+    new_pos: MaybePosition = pos.copy()  # becomes MaybePos later
     for tag, index in complete_offsets:
         new_pos = execute_shift(new_pos, tag, index, offset_provider=offset_provider)
         if new_pos is None:
             return None
     return new_pos
```

Comment:
I'm not sure, byt maybe `mypy` accepts the following alternative implementation:
```suggestion
    new_pos = pos.copy()
    for tag, index in complete_offsets:
        if (shifted_pos := execute_shift(new_pos, tag, index, offset_provider=offset_provider)) is not None:
            new_pos = shifted_pos
        else:
            return None
    return new_pos
```